docs/tasks/administer-cluster/encrypt-data.md

```diff
@@ -0,0 +1,205 @@
+
+
+## Verifying that data is encrypted 
+
+Data is encrypted when written to etcd. After restarting your `kube-apiserver`, any newly created or
+updated secret should be encrypted when stored. To check, you can use the `etcdctl` command line
+program to retrieve the contents of your secret.
+
+Create a new secret called `secret1` in the `default` namespace:
+
+```
+kubectl create secret generic secret1 -n default --from-literal=mykey=mydata
+```
+
+Using the etcdctl commandline, read that secret out of etcd:
+
+```
+ETCDCTL_API=3 etcdctl get /kubernetes.io/secrets/default/secret1 [...] | hexdump -C
+```
+
+where `[...]` must be the additional arguments for connecting to the etcd server. The `hexdump` command
+will format the encoded bytes into a readable form. Verify the resulting output is prefixed with 
+`k8s:enc:aescbc:v1:` which indicates the `aescbc` provider has encrypted the resulting data. 
+
+Verify the secret is correctly decoded by retrieving the secret from the command line, and that the secret
+contents match what was provided above (`mykey: mydata`).
+
+```
+kubectl describe secret generic -n default
+```
+
+
+## Ensuring all secrets to be encrypted
+
+Since secrets are encrypted on write, performing an update on a secret will encrypt that content.
+
+```
+kubectl get secrets -o json | kubectl update -f -
+```
+
+Will read all secrets and then perform an update with encryption. If an error occurs due to a 
+conflicting write, retry the command. For larger clusters, you may wish to subdivide the secrets
+by namespace or script an update.
+
+
+## Rotating a decryption key
+
+Changing the secret without incurring downtime requires a multi step operation, especially in
+the presence of a highly available deployment where multiple `kube-apiserver` processes are running.
+
+1. Generate a new key and add it as the second key entry for the current provider on all servers
+2. Restart all `kube-apiserver` processes to ensure each server can decrypt using the new key
+3. Make the new key the first entry in the `keys` array so that it is used for encryption in the config
+4. Restart all `kube-apiserver` processes to ensure each server now encrypts using the new key
+5. Run `kubectl get secrets -o json | kubectl update -f -` to update all secrets
+6. Remove the old decryption key from the config after you back up etcd with the new key in use and update all secrets
+
+With a single `kube-apiserver`, step 2 may be skipped
+
+
+## Decrypting all data
+
+To disable encryption at rest place the `identity` provider as the first entry in the config:
+
+```yaml
+kind: EncryptionConfig
+apiVersion: v1
+resources:
+  - resources:
+    - secrets
+    providers:
+    - identity: {}
+    - aescbc:
+        keys:
+        - name: key1
+          secret: <BASE 64 ENCODED SECRET>
+```
+
+and restart all `kube-apiserver` processes. Then run the command `kubectl get secrets -o json | kubectl update -f -`
+to force all secrets to be decrypted.
```

Comment:
You need to close this section w/ a `{% endcapture %}` tag and add the include for the Task template `{% include templates/task.md %}`, i.e.:

```
and restart all `kube-apiserver` processes. Then run the command `kubectl get secrets -o json | kubectl update -f -`
to force all secrets to be decrypted.
{% endcapture %}

{% include templates/task.md %}
```